pkg/artifactory/resource_artifactory_ldap_group_setting.go

```diff
@@ -0,0 +1,232 @@
+package artifactory
+
+import (
+	"context"
+	"encoding/xml"
+	"github.com/go-resty/resty/v2"
+	"gopkg.in/yaml.v2"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type LdapGroupSetting struct {
+	Name                 string `xml:"name" yaml:"name"`
+	EnabledLdap          string `xml:"enabledLdap" yaml:"enabledLdap"`
+	GroupBaseDn          string `xml:"groupBaseDn" yaml:"groupBaseDn"`
+	GroupNameAttribute   string `xml:"groupNameAttribute" yaml:"groupNameAttribute"`
+	GroupMemberAttribute string `xml:"groupMemberAttribute" yaml:"groupMemberAttribute"`
+	SubTree              bool   `xml:"subTree" yaml:"subTree"`
+	Filter               string `xml:"filter" yaml:"filter"`
+	DescriptionAttribute string `xml:"descriptionAttribute" yaml:"descriptionAttribute"`
+	Strategy             string `xml:"strategy" yaml:"strategy"`
+}
+
+type LdapGroupSettings struct {
+	LdapGroupSettingArr []LdapGroupSetting `xml:"ldapGroupSetting" yaml:"ldapGroupSetting"`
+}
+
+type SecurityLdapGroupSettings struct {
+	LdapGroupSettings LdapGroupSettings `xml:"ldapGroupSettings"`
+}
+
+type XmlLdapGroupConfig struct {
+	XMLName  xml.Name                  `xml:"config"`
+	Security SecurityLdapGroupSettings `xml:"security"`
+}
+
+func resourceArtifactoryLdapGroupSetting() *schema.Resource {
+	var ldapGroupSettingsSchema = map[string]*schema.Schema{
+		"name": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+			Description:      `(Required) Ldap group setting name.`,
+		},
+		"enabled_ldap": {
```

Comment:
BTW: This is weird - it reads like this should be a boolean

Reply:
changed the name to ldap_setting_key ... and added description to explain the link to the system configuration.